packages/amplify-category-function/src/provider-utils/awscloudformation/secrets/ssmClientWrapper.ts

```diff
@@ -93,15 +95,19 @@ export class SSMClientWrapper {
   };
 
   /**
-   * Delets all secrets in secretNames
+   * Deletes all secrets in secretNames
    */
-  deleteSecrets = async (secretNames: string[]) => {
+  deleteSecrets = async (secretNames: string[]): Promise<void> => {
     await this.ssmClient.deleteParameters({ Names: secretNames }).promise();
   };
 }
 
-const getSSMClient = async (context: $TSContext) => {
+const getSSMClient = async (context: $TSContext): Promise<aws.SSM> => {
   const spinner = ora('Initializing SSM Client');
+  if (context.exeInfo) {
+    context.exeInfo.spinner = spinner;
+  }
```

Comment:
I have moved the spinner to amplify-cli-core and you can use that instead of passing it through the context

Reply:
I'm not sure whether I can use this, since `amplify-cli-core` creates a new spinner every time but `context.exeInfo.spinner` is a singleton where I can check if it's already spinning or not.

Reply:
We want to reduce the dependency on the context object and the imported instance from the core is a singleton

Reply:
I might be misunderstanding but it doesn't look like a singleton here: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli-core/src/spinner/index.ts#L3